packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -50,11 +50,11 @@ class BuildInfo {
   /// Extra command-line options for gen_snapshot.
   final String extraGenSnapshotOptions;
 
-  /// Whether to prefer AOT compiling to a *so file.
-  final bool buildSharedLibrary;
+  // Whether to split the shared library per ABI.
+  final bool splitPerAbi;
 
-  /// Target platform for the build (e.g. android_arm versus android_arm64).
-  final TargetPlatform targetPlatform;
+  /// The target platforms for the build (e.g. android_arm or android_arm64).
```

Comment:
Can you document valid values?  e.g. is this guaranteed to be non-empty?  Or can it be empty, but that signals invalid build info?

Reply:
So, I ended up refactoring this code a little bit. Now, there's `AndroidBuildInfo`which contains a `BuildInfo`, `targetArchs` and `splitPerAbi`.   I defined `AndroidArch`, which mirrors `IosArch`.

Long term goal is to remove the concept of `android-arm`, `android-arm64`..., and instead use the abi values directly: https://developer.android.com/ndk/guides/abis

This makes the code self-explanatory and avoids combinations that aren't used in other target platforms. Indeed, the current `targetPlatform` in `BuildInfo` only makes sense in Android to pass arm, arm64 to gradle.